tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp

```diff
@@ -222,25 +222,24 @@ int main(int argc, char **argv) {
         initialize_program(program[1], fast_kernel_cycles_g);
 
         // Cache stuff
+        EnqueueProgram(cq, program[0], false);
+        EnqueueProgram(cq, program[1], false);
+        uint32_t tid = BeginTraceCapture(device, cq.id());
         for (int i = 0; i < warmup_iterations_g; i++) {
             EnqueueProgram(cq, program[0], false);
             if (nfast_kernels_g > 0) {
                 EnqueueProgram(cq, program[1], false);
             }
         }
+        EndTraceCapture(device, cq.id(), tid);
         Finish(cq);
 
         if (lazy_g) {
             tt_metal::detail::SetLazyCommandQueueMode(true);
         }
 
         auto start = std::chrono::system_clock::now();
-        for (int i = 0; i < iterations_g; i++) {
-            EnqueueProgram(cq, program[0], false);
-            for (int j = 0; j < nfast_kernels_g; j++) {
-                EnqueueProgram(cq, program[1], false);
-            }
-        }
+        EnqueueTrace(cq, tid, true);
```

Comment:
We don't need a blocking call here if we call `Finish` right after. I suggest setting `blocking = false`, otherwise we'll be profiling additional host overhead: a finish command being generated and sent at the end of the trace and another one when we call `Finish`.

Reply:
Sounds good. Will make change